__test__/integration/adapters/collateralized_simple_interest_loan_adapter/runners/return_collateral_runner.ts

```diff
@@ -0,0 +1,301 @@
+import { CollateralizedSimpleInterestTermsContractContract } from "src/wrappers/contract_wrappers/collateralized_simple_interest_terms_contract_wrapper";
+import { TokenTransferProxyContract } from "src/wrappers/contract_wrappers/token_transfer_proxy_wrapper";
+
+// APIs
+import { OrderAPI } from "src/apis/order_api";
+import { SignerAPI } from "src/apis/signer_api";
+import { ServicingAPI } from "src/apis/servicing_api";
+
+// Types
+import { DebtOrder } from "src/types/debt_order";
+
+// Accounts
+import { ACCOUNTS } from "__test__/accounts";
+
+const CONTRACT_OWNER = ACCOUNTS[0];
+const DEBTOR = ACCOUNTS[1];
+const CREDITOR = ACCOUNTS[2];
+const UNDERWRITER = ACCOUNTS[3];
+
+// A list of contract wrappers that are required for the following test examples.
+export interface ContractWrappers {
+    debtKernel: DebtKernelContract;
+    repaymentRouter: RepaymentRouterContract;
+    principalToken: DummyTokenContract;
+    collateralToken: DummyTokenContract;
+    termsContract: CollateralizedSimpleInterestTermsContractContract;
+}
+
+export interface APIs {
+    orderApi: OrderAPI;
+    signerApi: SignerAPI;
+    servicingApi: ServicingAPI;
+}
+
+export class ReturnCollateralRunner {
+    private adapter: CollateralizedSimpleInterestLoanAdapter;
+    private debtKernel: DebtKernelContract;
+    private repaymentRouter: RepaymentRouterContract;
+    private principalToken: DummyTokenContract;
+    private collateralToken: DummyTokenContract;
+    private termsContract: CollateralizedSimpleInterestTermsContractContract;
+    private orderApi: OrderAPI;
+    private signerApi: SignerAPI;
+    private tokenTransferProxy: TokenTransferProxyContract;
+    private web3Utils: Web3Utils;
+    private servicingApi: ServicingAPI;
+    private snapshotId: number;
+    private debtOrder: DebtOrder.Instance;
+
+    constructor() {
+        this.testScenario = this.testScenario.bind(this);
+        this.revertToSavedSnapshot = this.revertToSavedSnapshot.bind(this);
+    }
+
+    public async saveSnapshotAsync() {
+        this.snapshotId = await this.web3Utils.saveTestSnapshot();
+    }
+
+    public async revertToSavedSnapshot() {
+        await this.web3Utils.revertToSnapshot(this.snapshotId);
+    }
+
+    public initialize(
+        web3: Web3,
+        adapter: CollateralizedSimpleInterestLoanAdapter,
+        tokenTransferProxy: TokenTransferProxyContract,
+        contractWrappers: ContractWrappers,
+        apis: APIs,
+    ) {
+        this.debtKernel = contractWrappers.debtKernel;
+        this.termsContract = contractWrappers.termsContract;
+        this.repaymentRouter = contractWrappers.repaymentRouter;
+        this.principalToken = contractWrappers.principalToken;
+        this.collateralToken = contractWrappers.collateralToken;
+
+        this.orderApi = apis.orderApi;
+        this.signerApi = apis.signerApi;
+        this.servicingApi = apis.servicingApi;
+
+        this.adapter = adapter;
```

Comment:
I think we can obviate the need for this `initialize` method -- all of the above contracts can be loaded in the `beforeAll` block using either the APIs or (in the case of the DummyTokens) their contract wrappers.  Please refactor to remove the dependency on running `initialize` -- it's much less brittle to only rely on the constructor as an initializer.